pkg/utils/selector.go

```diff
@@ -119,9 +119,13 @@ func SelectPods(ctx context.Context, c client.Client, selector v1alpha1.Selector
 	}
 	if len(selector.FieldSelectors) > 0 {
 		listOptions.FieldSelector = fields.SelectorFromSet(selector.FieldSelectors)
-	}
-	if err := c.List(ctx, &podList, &listOptions); err != nil {
-		return nil, err
+		if err := r.List(ctx, &podList, &listOptions); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := c.List(ctx, &podList, &listOptions); err != nil {
+			return nil, err
+		}
```

Comment:
how about adding a comment about why in this case use client.Reader and in another case use client.Client?